loader/loader.go

```diff
@@ -0,0 +1,55 @@
+package loader
+
+import (
+	"context"
+	"github.com/redhat-appstudio/internal-services/api/v1alpha1"
+	"github.com/redhat-appstudio/internal-services/tekton"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ObjectLoader interface {
+	GetInternalRequest(ctx context.Context, cli client.Client, name, namespace string) (*v1alpha1.InternalRequest, error)
+	GetInternalRequestPipelineRun(ctx context.Context, cli client.Client, release *v1alpha1.InternalRequest) (*v1beta1.PipelineRun, error)
+}
+
+type loader struct{}
+
+func NewLoader() ObjectLoader {
+	return &loader{}
+}
+
+// getObject loads an object from the cluster. This is a generic function that requires the object to be passed as an
+// argument. The object is modified during the invocation.
+func getObject(name, namespace string, cli client.Client, ctx context.Context, object client.Object) error {
```

Comment:
I really think this should be in the goodies repo and should not exist in our service repos

Reply:
I thought about it, but we will have to revisit once the Loader if fully implemented.